### PR TITLE
Desktop 1774 part 1

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Upgrades us to flow v0.32! a precursor to a better typed connect

### CODE REVIEW

File: shared/login/register/error/index.render.desktop.js [hunk 1]
```diff
@@ -8,8 +8,9 @@ import {ConstantsStatusCode} from '../../../constants/types/flow-types'
 import {Text} from '../../../common-adapters'
 
 const renderError = (error: RPCError) => {
-  const fields = (error.fields || []).reduce((acc, f) => {
-    acc[f.key] = f.value
+  const fields = (Array.isArray(error.fields) ? error.fields : []).reduce((acc, f) => {
+    const k = f && typeof f.key === 'string' ? f.key : ''
```
Comment (chromakode) [inline shared/login/register/error/index.render.desktop.js]: :sheep: This is a little wonky (falling back to the key of `''`) but probably not worth complicating it more. Flow isn't really helping here :/

File: shared/menubar/index.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ class Menubar extends Component<void, Props, void> {
       })
     }
 
-    if (module.hot) {
+    if (typeof module.hot === 'function') {
```
Comment (chromakode) [inline shared/menubar/index.js]: I believe `module.hot` should be an object at runtime. Just checked and it appears so.

Comment (MarcoPolo) [inline shared/menubar/index.js]: ah whoops! was just quickly fixing errors, thanks!

File: shared/login/register/error/index.render.desktop.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/login/register/error/index.render.desktop.js]: it was falling back to undefined before

**Comment by chromakode** [comment]
:+1: w/ `module.hot` fixed

**Comment by chrisnojima** [comment]
why not use ```{| |}```?

**Comment by MarcoPolo** [comment]
@chrisnojima Flow's new support for Exact types is different than our version of Exact types. It's lacking a couple things we use and our Exact types work for (https://github.com/facebook/flow/issues/2405) So this change renamed our $Exact (which conflicts with Flow's new exact support) to just `Exact`.


Besides that reason for using our exact, our eslint flow plugin does not like `{| |}` yet.

**Comment by chromakode** [comment]
:+1:

**Comment by cjb** [comment]
@MarcoPolo wonder why this didn't merge in?

**Comment by MarcoPolo** [comment]
@cjb I was waiting for part 2: https://github.com/keybase/client/pull/4273

**Pull request merged**